### PR TITLE
feat(retry): allow respecting Retry-After for user-defined status codes

### LINKED ISSUE #473: Using "Retry-After" with error code 425 is impossible
Hi there,

I would like to change the way the retries are made. For example: wait 1s before each retry.

One option is to [add a `retry.backoffLimit` option](https://github.com/sindresorhus/ky/pull/454)
Another (by the time the PR is merged) would be to use the `Retry-After` header from the server.
But unfortunately I can make it work with the `425` error code given by the server.
It looks like the `afterStatusCodes` option, set by default to `[413, 429, 503]` cannot be changed.
I've tried these retry options :
```
retry: {
  limit: 12,
  afterStatusCodes: [425],
  statusCodes: [425],
}
```
The code I'm looking at is here: [(Ky.ts) _calculateRetryDelay l.217](https://github.com/sindresorhus/ky/blob/430486b5522dcafd2d658ab391c6a37d921d66da/source/core/Ky.ts#L217)

`afterStatusCodes` is not in the docs… but I found it reading the code and in [that article](https://golb.hplar.ch/2019/08/ky.html)

Any help is welcome

### PR DESCRIPTION
**Pull request by Kenneth-Sills**
Implemented as recommended in related issue, with appropriate testing and documentation.

The included test is just a copy of the existing 413 one, but with a 500 and required config instead.

Closes #473.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -212,14 +212,17 @@ Default:
...
 
-An object representing `limit`, `methods`, `statusCodes` and `maxRetryAfter` fields for maximum retry count, allowed methods, allowed status codes and maximum [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) time.
+An object representing `limit`, `methods`, `statusCodes`, `afterStatusCodes`, and `maxRetryAfter` fields for maximum retry count, allowed methods, allowed status codes, status codes allowed to use the [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) time, and maximum [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) time.
 
 If `retry` is a number, it will be used as `limit` and other defaults will remain in place.
 
+If the response provides an HTTP status contained in `afterStatusCodes`, Ky will wait until the date or timeout given in the [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) header has passed to retry the request. If the provided status code is not in the list, the [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) header will be ignored (retries sent immediately).
```
Comment (sindresorhus) [inline readme.md:224]: > retries sent immediately

That really depends on the retry configuration, so I would drop this part.

**Comment by sindresorhus** [comment]
You also need to update https://github.com/sindresorhus/ky/blob/e044626ad8163f00a1fd3834558c1a13d5c2b2de/source/types/options.ts#L118-L142

**Comment by Kenneth-Sills** [comment]
Thanks for the review! I haven't had a lot of free time this past week, but I hope to get to this soon. Feel free to bump me if it gets stale, I do check notifications.

File: test/retry.ts [hunk 2]
```diff
@@ -5,7 +5,9 @@ import {withPerformanceObserver} from './helpers/with-performance-observer.js';
 
 const fixture = 'fixture';
 const defaultRetryCount = 2;
+const retryAfterOn500 = 2;
 const retryAfterOn413 = 2;
+const lastTried500access = Date.now();
```
[the remark this answers is not in the document]
Comment (Kenneth-Sills) [reply test/retry.ts:10]: Yeah, to be honest I had tunnel vision and barely paid attention beyond getting it working with the existing paradigm.

**Pull request merged**